google/cloud/forseti/scanner/scanners/data_models/data_model_builder.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2017 The Forseti Security Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Builds the data models."""
+
+from builtins import object
+import importlib
+
+from google.cloud.forseti.common.util import logger
+from google.cloud.forseti.scanner.scanners.data_models \
+    import data_model_requirements_map
+
+LOGGER = logger.get_logger(__name__)
+
+
+class DataModelBuilder(object):
+    """Data Model Builder."""
+    def __init__(self, global_configs, service_config,
+                 model_name):
+        """Initialize the data model builder.
+
+        Args:
+            global_configs (dict): Global configurations.
+            service_config (ServiceConfig): Service configuration.
+            model_name (str): name of the data model
+        """
+        self.global_configs = global_configs
+        self.service_config = service_config
+        self.model_name = model_name
+
+    def build(self):
+        """Build the data models.
+
+        Returns:
+            list: data model instances that will be created.
+        """
+        data_models = []
+        for data_model in data_model_requirements_map.REQUIREMENTS_MAP:
```

Comment:
Will there be any logic that will only build the data model that are needed?  Say, what if someone disable the IAP scanner?

Reply:
That is a good point- maybe it could be similar to the scanners where users can choose to enable/disable certain resources?

Reply:
Have added the scanner_configs parameter so that if a scanner is disabled, the data model for that scanner would not be created.